nautobot/extras/tests/integration/test_plugins.py

```diff
@@ -0,0 +1,107 @@
+import os
+import re
+from unittest import skipIf
+
+from django.conf import settings
+from django.contrib.contenttypes.models import ContentType
+from django.urls import reverse
+
+
+from nautobot.extras.choices import WebhookHttpMethodChoices
+from nautobot.extras.context_managers import web_request_context
+from nautobot.extras.models import Webhook
+from nautobot.users.models import Token
+from nautobot.utilities.testing.integration import SplinterTestCase
+
+from dummy_plugin.models import DummyModel
+
+
+@skipIf(
+    "dummy_plugin" not in settings.PLUGINS,
+    "dummy_plugin not in settings.PLUGINS",
+)
+class PluginWebhookTest(SplinterTestCase):
+    """
+    This test case proves that plugins can use the webhook functions when making changes on a model.
+
+    Because webhooks use celery a class variable is set to True called `requires_celery`. This starts
+    a celery instance in a separate thread.
+    """
+
+    requires_celery = True
+
+    def setUp(self):
+        super().setUp()
+        for f in os.listdir("/tmp"):
+            if re.search("test_plugin_webhook_.+", f):
+                os.remove(os.path("/tmp", f))
+
+        self.url = f"http://localhost:{self.server_thread.port}" + reverse(
+            "plugins-api:dummy_plugin-api:dummymodel_webhook"
+        )
+        self.login(self.user.username, self.password)
+        self.token = Token.objects.create(user=self.user)
+        self.header = f"Authorization: Token {self.token.key}"
+        self.webhook = Webhook.objects.create(
+            name="DummyModel",
+            type_create=True,
+            type_update=True,
+            type_delete=True,
+            payload_url=self.url,
+            http_method=WebhookHttpMethodChoices.METHOD_GET,
+            additional_headers=self.header,
+        )
+        dummy_ct = ContentType.objects.get_for_model(DummyModel)
+        self.webhook.content_types.set([dummy_ct])
+
+    def update_headers(self, new_header):
+        """
+        Update webhook additional headers with the name of the running test.
+        """
+        headers = self.header + f"\nTest-Name: {new_header}"
+        self.webhook.additional_headers = headers
+        self.webhook.validated_save()
+
+    def test_plugin_webhook_create(self):
+        """
+        Test `process_webhook` from a model create.
+        """
+        self.update_headers("test_plugin_webhook_create")
+        self.clear_worker()
+        # Make change to model
+        with web_request_context(self.user):
+            DummyModel.objects.create(name="foo", number=100)
+        self.wait_on_active_tasks()
+        self.assertTrue(os.path.exists("/tmp/test_plugin_webhook_create"))
+        os.remove("/tmp/test_plugin_webhook_create")
+
+    def test_plugin_webhook_update(self):
+        """
+        Test `process_webhook` from a model update.
+        """
+        self.update_headers("test_plugin_webhook_update")
```

Comment:
Maybe paranoia on my part, but I'd consider calling `self.update_headers` *after* `self.clear_worker()` just to make sure that the webhook with those headers is correctly being sent at the expected time only, and no earlier.